src/main/java/com/epam/reportportal/junit/ParallelRunningHandler.java

```diff
@@ -307,11 +301,38 @@ protected StartTestItemRQ buildStartStepRq(FrameworkMethod method) {
 		rq.setUniqueId(extractUniqueID(method));
 		rq.setStartTime(Calendar.getInstance().getTime());
 		rq.setType(detectMethodType(method));
+		rq.setTags(getAnnotationTags(method));
 
 		rq.setRetry(isRetry(method));
 		return rq;
 	}
 
+	/**
+	 * Returns a set of tags of the given annotated context. This method is looking for the following annotations in
+	 * this context:
+	 *
+	 * <ul>
+	 * <li>{@link Tags}</li>
+	 * <li>{@link Category}</li>
+	 * </ul>
+	 * <p>
+	 * If annotations are not present - this method returns empty set.
+	 *
+	 * @param context Context to extract tags
+	 * @return Set of tags of given annotated element
+	 */
+	private Set<String> getAnnotationTags(Annotatable context) {
+		Set<String> result = new HashSet<>();

```

Comment:
no need to create new set before actual logic. 
Just use .orElse(Collections.emptySet) when you need it